src/azure-cli/azure/cli/command_modules/resource/commands.py

```diff
@@ -233,19 +234,86 @@ def load_command_table(self, _):
-        g.custom_command('create', 'deploy_arm_template_at_subscription_scope', supports_no_wait=True, validator=process_deployment_create_namespace, exception_handler=handle_template_based_exception)
-        g.custom_command('export', 'export_subscription_deployment_template')
-        g.custom_wait_command('wait', 'wait_deployment_at_subscription_scope')
+        g.custom_command('validate', 'validate_arm_template_at_subscription_scope', validator=process_deployment_create_namespace,
+                         table_transformer=deployment_validate_table_format, exception_handler=handle_template_based_exception)
+        g.custom_command('create', 'deploy_arm_template_at_subscription_scope', supports_no_wait=True, validator=process_deployment_create_namespace,
+                         exception_handler=handle_template_based_exception)
+        g.custom_command('export', 'export_template_at_subscription_scope')
+        g.custom_wait_command('wait', 'get_deployment_at_subscription_scope')
+
+    with self.command_group('deployment operation', resource_deployment_operation_sdk, min_api='2019-07-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES) as g:
+        g.custom_command('list', 'list_deployment_operations_at_subscription_scope')
+        g.custom_show_command('show', 'get_deployment_operations_at_subscription_scope', client_factory=cf_deployment_operations)
+
+    # az deployment sub
+    with self.command_group('deployment sub', resource_deployment_sdk, min_api='2019-07-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES) as g:
+        g.custom_command('list', 'list_deployments_at_subscription_scope', table_transformer=transform_deployments_list)
+        g.custom_show_command('show', 'get_deployment_at_subscription_scope', table_transformer=transform_deployment)
+        g.custom_command('delete', 'delete_deployment_at_subscription_scope', supports_no_wait=True)
+        g.custom_command('validate', 'validate_arm_template_at_subscription_scope', validator=process_deployment_create_namespace,
+                         table_transformer=deployment_validate_table_format, exception_handler=handle_template_based_exception)
+        g.custom_command('create', 'deploy_arm_template_at_subscription_scope', supports_no_wait=True, validator=process_deployment_create_namespace,
+                         table_transformer=transform_deployment, exception_handler=handle_template_based_exception)
+        g.custom_command('export', 'export_template_at_subscription_scope')
+        g.custom_wait_command('wait', 'get_deployment_at_subscription_scope')
 
-    with self.command_group('deployment operation', resource_deployment_operation_sdk, min_api='2018-05-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES) as g:
+    with self.command_group('deployment operation sub', resource_deployment_operation_sdk, min_api='2019-07-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES) as g:
```

Comment:
It looks the commands in `az deployment operation sub` commands group are the same with commands in `az deployment operation`. https://github.com/Azure/azure-cli/pull/10751/files?file-filters%5B%5D=.json&file-filters%5B%5D=.py#diff-2057bbb8441166e4910b34b09d22b58cR249

Reply:
right. that's expected. The command group `az deployment operation sub` is the same as `az deployment operation`, and `az deployment sub` is the same as `az deployment`. We want to retire `az deployment` and `az deployment operation` but guess we'll do that in the future.

Reply:
makes sense. will fix it.

Reply:
If you will deprecate some commands group, please add deprecate info to them as shown in https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments

Reply:
may I understand why az deployment operation will be retired? is there more subresource like sub in operation command group, if so, can you give an example? I ask this because if sub is already the top resource in operation, we would suggest to have it as a default resource in operation command group.

---
In reply to: [367718352](https://github.com/Azure/azure-cli/pull/10751#discussion_r367718352) [](ancestors = 367718352)

Reply:
so `az deployment operation` works at subscription scope today. It lists or shows the deployment operations for a deployment at subscription scope. With the changes in this PR, we are supporting deployment at tenant and management group scope. To make all commands consistent, we are introducing `az deployment operation group | sub | mg | tenant` for each scope. The `az deployment operation sub` will replace `az deployment operation`.